### PR TITLE
Add lsp-start-plain.el to provide a quick starter for minimal env.

### PR DESCRIPTION
**Pull request by cireu**
See https://github.com/emacs-lsp/lsp-mode/issues/592

How to use this:
1. Download the file and run it with `emacs -q`
2. using `emacs -q --eval '(with-current-buffer (url-retrieve-synchronously "http://github.com/cireu/lsp-mode/raw/master/lsp-start-plain.el") (goto-char (point-max)) (eval-print-last-sexp))'

I've test it using [emacs-china](http://elpa.emacs-china.org/) elpa mirror. But my network is too slow to connect to melpa. Can someone help testing it?

### CODE REVIEW

**Comment by cireu** [comment]
I still can't find a way to avoid recursive loading issue. But I find it's possible to let `package.el` don't compile packages.

So I rewrote it. Can someone help test it on Windows and MacOS platform? Thx a lot :smile:

File: scripts/lsp-start-plain.el [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+;; for more readable backtrace.
+
+;;; Code:
+
+
+(let* ((lexical-binding t)
+       (package-archives '(("melpa" . "http://melpa.org/packages/")
+                           ("gnu" . "http://elpa.gnu.org/packages/")))
+       (no-byte-compile t)
+       (package-user-dir (expand-file-name "tmpelpa" user-emacs-directory))
```
Comment (yyoncho) [inline scripts/lsp-start-plain.el]: maybe use make-directory/make-temp-name to generate new temp directory each time.

File: scripts/lsp-start-plain.el [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+;; It will install  `lsp-mode', `lsp-ui' and `company-lsp'
+;; with their depedencies to start a minimal lsp environment.
+
+;; And it forces Emacs to load `.el' files rather than `.elc' files
+;; for more readable backtrace.
+
+;;; Code:
+
+
+(let* ((lexical-binding t)
```
Comment (yyoncho) [inline scripts/lsp-start-plain.el]: not needed when headers starts with:
```
;;; lsp-start-plain.el --- LSP mode quick starter      -*- lexical-binding: t; -*-
```

**Comment by cireu** [comment]
I am considering to put this util function into a part of `lsp-mode` like magit did.

https://github.com/magit/magit/blob/cc435005b07bd7ba17962ffa4e210b441e8a1a52/lisp/magit-utils.el#L725

What's your opinion? Keep it a single file or merge it into `lsp-mode`?

**Comment by cireu** [comment]
Er.. Can you describe your idea with more details? In my understanding, Cask has done what you said.

IMO, we can get this PR done first, so users can use it and benefit from it. And then is time to think about what to do next. We may need more people's help and idea.

**Comment by yyoncho** [comment]
My idea which is not directly related to that issue is to create one meta package, let's call it `lsp-ide` which will have a dependency to all lsp related - lsp-mode/lsp-ui/company-lsp/helm-lsp + few more that I am planning in the future. What I am saying is that a method like `lsp-ide-emacs-Q` looks like a good fit for that package.

**Comment by cireu** [comment]
Ah.. You mean a package(`lsp-ide`) to intergrate and associate these packages to create a `lsp-mode` based workflow in Emacs (like [aweshell](https://github.com/manateelazycat/aweshell), lazycat packs some package to provide an out-of-box and fancy eshell)? That sounds good! And such `lsp-ide-emacs-Q` functions will be more suitable in `lsp-ide` than `lsp-mode`

**Comment by cireu** [comment]
My opinion is to keep this file single now and let it be a part of `lsp-ide` in near future :)

**Pull request merged**